Automattic/jetmon
# Update: Add monitor_url to verflier messages

The verifliers code was not fully updated to handle sending the `monitor_url` along with the `blog_id`. This means that the verifliers could not accurately report which site they were reporting results for. This PR is a partial fix for this issue. This fix focuses on ensuring that the `monitor_url` is passed through the messaging sent back to Jetmon.

A big issue remains however: the retry queue tracks sites to retry by just the `blog_id`. This means that the retry queue is ambiguous. The entire retry queue system (including in the verifliers) must be updated to a system that either utilizes the `jetpack_monitor_site_id` for unique site IDs or to somehow combine the `blog_id` and `monitor_id`. I believe that the first option, using `jetpack_monitor_site_id`, is a more simple approach, but everything in Jetmon, the workers, and the verifliers will all need to be updated to use this approach. If this approach is used, this PR is not necessary as the new ID will be sufficient to track both the `blog_id` and `monitor_url`.

## Testing Instruction

1. Checkout the latest `master` branch.
2. Add a `console.log` or similar debug output to the top of the `checkHostStatus` function in `lib/jetmon.js`, such as: `console.log({data: data});`
3. Run Jetmon and have at least one test site change its status.
4. Confirm that the debug output shows that the `data` object does not have an entry for `monitor_url`.
5. Checkout out this PR's branch (`update/add-monitor_url-to-veriflier-messages`).
6. Rebuild the project: `sudo docker compose build --no-cache`
7. Ensure that the same debug output code as you added in step 2 is present.
8. Run Jetmon and have at least one test site change its status.
9. Confirm that the debug output shows that the `data` object now has an entry for `monitor_url`.
